rewrite-yaml/src/test/java/org/openrewrite/yaml/tree/MappingTest.java

```diff
@@ -408,6 +448,27 @@ void mappingAnchor() {
         );
     }
 
+    @Test
+    void mappingSequenceWithAnchor() {
+        rewriteRun(
+          yaml(
+            """
+              defaults:
+                - &first
+                  A: 1
+                  B: 2
+              mapping:
+                - &first
+                  A: 1
+                  B: 2
+                - <<: *first
+                  A: 23
+                  C: 99
+              """
+          )
+        );
+    }
```

Comment:
In case this is helpful for review… here's a screenshot of debugging this test, showing  <kbd>Sequence</kbd> -> <kbd>Sequence.Entry</kbd> -> <kbd>Mapping</kbd> with anchor and entries in the expected location. Expand the image to show the anchor prefix is the space between the `-` and `&`. I hope this is acceptable.

<img width="896" height="631" alt="image" src="https://github.com/user-attachments/assets/700350ff-41b0-445c-80ca-f413b508021d" />